src/game/enemy/systems.rs

```diff
@@ -1 +1,172 @@
-use bevy::prelude::*;
\ No newline at end of file
+use bevy::{prelude::*, window::PrimaryWindow};
+
+use crate::components::{AnimationIndices, AnimationTimer};
+use crate::game::enemy::components::*;
+use crate::game::SpriteSheetInfo;
+
+// https://github.com/bevyengine/bevy/blob/main/examples/2d/sprite_sheet.rs
+
+const SOLDIER_01_IDLE: SpriteSheetInfo = SpriteSheetInfo {
+    path: "sprites/soldier_01/Idle.png",
+    x: 128.0,
+    y: 128.0,
+    cols: 7,
+    rows: 1,
+};
+const SOLDIER_01_RUN: SpriteSheetInfo = SpriteSheetInfo {
+    path: "sprites/soldier_01/Run.png",
+    x: 128.0,
+    y: 128.0,
+    cols: 7,
+    rows: 1,
+};
+
+// Sometimes the edges are white. Possible issues: z-index fighting, need background
+pub fn animate_sprite(
```

Comment:
In theory we would have sprite animation for allies too, and perhaps other things like fire. So maybe this would be worth its own component?

Reply:
I tried to implement it in a way that it could be shared, but because of the query system params I couldn't figure it out. Idk if it can be written it in a general way that works for all sprites

Reply:
Couldnt we just have a generic "Animated" struct attached to the sprite, and query all with those?